crates/sui-node/src/lib.rs

```diff
@@ -242,18 +242,23 @@ impl SuiNode {
         // Create network
         // TODO only configure validators as seed/preferred peers for validators and not for
         // fullnodes once we've had a chance to re-work fullnode configuration generation.
-        let (trusted_peer_change_tx, trusted_peer_change_rx) =
-            watch::channel(TrustedPeerChangeEvent {
-                new_peers: epoch_store
-                    .epoch_start_state()
-                    .get_validator_as_p2p_peers(config.protocol_public_key()),
-            });
+        let (trusted_peer_change_tx, trusted_peer_change_rx) = broadcast::channel(100);
         let (p2p_network, discovery_handle, state_sync_handle) = Self::create_p2p_network(
             &config,
             state_sync_store,
             trusted_peer_change_rx,
             &prometheus_registry,
         )?;
+        if let Err(err) = trusted_peer_change_tx.send(TrustedPeerChangeEvent {
```

Comment:
Explicitly sending over the list of `TrustedPeerChangeEvent` could be the difference.